contracts/InterchainTokenService.sol

```diff
@@ -332,27 +360,29 @@ contract InterchainTokenService is
         // Custom token managers can't be deployed with native interchain token type, which is reserved for interchain tokens
         if (tokenManagerType == TokenManagerType.NATIVE_INTERCHAIN_TOKEN) revert CannotDeploy(tokenManagerType);
 
+        // Cannot deploy to this chain using linkToken anymore
+        if (bytes(destinationChain).length == 0) {
+            revert NotSupported();
+        }
+
+        // Cannot deploy to this chain using linkToken anymore
+        if (chainNameHash == keccak256(bytes(destinationChain))) revert CannotDeployRemotelyToSelf();
+
         address deployer = msg.sender;
 
         if (deployer == interchainTokenFactory) {
             deployer = TOKEN_FACTORY_DEPLOYER;
-        } else if (bytes(destinationChain).length == 0) {
-            // TODO: Only support linking new tokens via ITS factory, to include chain name in token id derivation
-            // Custom token usage needs to be moved to ITS factory tests
-            // revert NotSupported();
         }
 
         tokenId = interchainTokenId(deployer, salt);
 
-        emit InterchainTokenIdClaimed(tokenId, deployer, salt);
+        // Make sure that the token manager already exists on this chain.
+        deployedTokenManager(tokenId);
```

Comment:
`_linkToken` already checks this by getting source token address. Perhaps `_linkToken` code can be merged with this method now

Reply:
nice catch, removed